src/main/scala/net/cakesolutions/CakeBuildInfoPlugin.scala

```diff
@@ -4,30 +4,45 @@ package net.cakesolutions
 
 import scala.util._
 
-import sbt._
-import sbt.IO._
 import sbt.Keys._
+import sbt._
+import sbtbuildinfo.BuildInfoPlugin
+import sbtbuildinfo.BuildInfoPlugin.autoImport._
 
-import sbtbuildinfo.BuildInfoPlugin, BuildInfoPlugin.autoImport._
-
-// kicks in if the Project uses the BuildInfoPlugin
+/**
+ * Assumes that `git` is installed. Used to configure and load the `BuildInfo`
+ * object at runtime. This object contains information regarding the build
+ * such as the git commit hash, etc.
+ *
+ * Configuration of this plugin should be avoided in local project SBT build
+ * files.
+ */
 object CakeBuildInfoPlugin extends AutoPlugin {
-  override def requires = BuildInfoPlugin
-  override def trigger = allRequirements
+  require(
```

Comment:
(this may be coming from a dependent PR) I'm not keen on `require` in the block of an `object`. If you want a check, put it in a `Task` somewhere so it will reliably fail.